src/arch/fdselecthand.cpp

```diff
@@ -131,8 +131,8 @@ void CFDSelectHandler::removeComCallback(TFileDescriptor paFD) {
 CFDSelectHandler::TFileDescriptor CFDSelectHandler::createFDSet(fd_set *mFDSet) {
   TFileDescriptor nRetVal = scmInvalidFileDescriptor;
   FD_ZERO(mFDSet);
-  TConnectionContainer::Iterator itEnd(mConnectionsList.end());
-  for (TConnectionContainer::Iterator itRunner = mConnectionsList.begin(); itRunner != itEnd; ++itRunner) {
```

Comment:
please use the more efficient for each loop:

for(auto it : mConnecitonList) {... }